### ISSUE TITLE
function closure variables with adaptive entry point

### ISSUE THREAD

**Issue by toth-istvan-zoltan** [spxbhuhb/adaptive#1]
This results in a compiler error:

```kotlin
fun a() {
    val i = 12
    adaptive(TestAdapter()) {
        T1(i)
    }
}
```

This works OK:

```kotlin
val i1 = 12
fun a() {
    adaptive(TestAdapter()) {
        T1(i1)
    }
}
```

The problem is the function scope which is something complicated I think as there may be other, non-adaptive players who change the value of the variable. 

Maybe the best solution would be to generate a compilation error in this case.

**Comment by toth-istvan-zoltan** [comment]
This actually works now. I've probably fixed it somewhere on the way.

Added a unit test to the compiler plugin tests for this.

**Comment by toth-istvan-zoltan** [comment]
Re-opening as 2.1.0 upgrade broke it.

Producer test cases in auto are affected by this.

It is a low priority issue as this is an uncommon pattern, useful mostly for unit tests.

**Comment by toth-istvan-zoltan** [comment]
`adaptive-kotlin-plugin/testData/box/foundation/variables/closure.kt`

Commented out the test for now, it fails.

**Comment by toth-istvan-zoltan** [comment]
Fixed in #89 .
